_layouts/base.html

```diff
@@ -4,6 +4,8 @@
   <meta charset="utf-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
 
+  <link rel="canoncial" href="{{ page.url | remove:'index.html' | remove:'.html' | absolute_url }}">
```

Comment:
I'm not sure what you're doing here, but maybe the permalink front-matter I referenced below might help.

https://jekyllrb.com/docs/permalinks/